src/main/java/org/cristalise/lookup/ldap/LDAPLookup.java

```diff
@@ -435,13 +435,16 @@ public ItemPath getItemPath(String uuid) throws ObjectNotFoundException, Invalid
         try {
         	ItemPath item = new ItemPath(uuid);
             LDAPEntry anEntry=mLDAPAuth.getAuthObject().read(getDN(item)+mLocalPath,attr);
-            String type = LDAPLookupUtils.getFirstAttributeValue(anEntry, "objectClass");
-            if (type.equals("cristalentity"))
-                return item;
-            else if (type.equals("cristalagent"))
-                return new AgentPath(item);
-            else
-                throw new ObjectNotFoundException("Not an entity '"+uuid+"'");
+            String[] values = LDAPLookupUtils.getAllAttributeValues(anEntry, "objectclass");
+            if (values != null) {
+            	for (String type : values) {
+            		if (type.equals("cristalentity"))
+                        return item;
```

Comment:
In the OpenLDAP schema, the cristalagent is declared like this:

`objectclass ( 1.3.6.1.4.1.96.64.10.14 NAME 'cristalagent' SUP cristalentity STRUCTURAL`

This may have the effect in some LDAP servers that agents will have both 'cristalagent' AND 'cristalentity' set in objectclass attributes. Even if ApacheDS doesn't do it now, it may later. It might be better to look at all of the attributes instead of returning entity when it's found.